wordpress-mobile/WordPress-Android
# Feature/15234 domains dashboard

Fixes #15234 

Adds **Domains** entry point on Site dashboard and adds a new **domains dashboard**

<img width=320 src="https://user-images.githubusercontent.com/990349/130933051-68e9251b-bb6e-4a2e-a1ab-3c9610358f18.png" />


To test:

1. Turn on **SiteDomainsFeatureConfig** and  **MySiteImprovementsFeatureConfig** from App Settings -> Debug settings 
2. Find new Domains item under Configuration on My Site dashboard
3. Tapping on Domains should launch Site domains dashboard with temporary UI
4. Click on Get your domain to launch Register Domain 

## Regression Notes
1. Potential unintended areas of impact
Site dashboard

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Unit testing

PR submission checklist:
